### ISSUE TITLE
Enhancement: return-await should allow awaiting on non promise values

### ISSUE THREAD

**Issue by matthieusieben** [typescript-eslint/typescript-eslint#5118]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/return-await/

### Description

This is related to #5117 but not exactly the same.

When a function's definition changes from being asynchronous to being synchronous (eg. due to a refactor, or invalid type definitions), it does not really hurt to `return await someValueThatIsNowSync`. There might be a performance issue when doing so but I believe we should be able to allow that, especially when using the `always` option.

### Fail

```typescript
// Not relevant
```

### Pass

```typescript
const foo = () => 42
const bar = async () => {
  try {
    return await foo()
  } catch {
    return 42
  }
}
```

**Comment by JoshuaKGoldberg** [comment]
> it does not really hurt to `return await someValueThatIsNowSync`

![giphy (4)](https://user-images.githubusercontent.com/3335181/171194297-d344983c-d067-4493-a0df-ceffea5d2561.gif)

...define "hurt"? It's unnecessary, potentially confusing extra code. If you know the value isn't a Thenable then there's no real reason to await on it.

I'm hesitant to add in a rule option for a case that is not as preferable. Are there strong points in favor of why we should allow this?

**Comment by bradzacher** [comment]
It does actually hurt to `await` unnecessarily!
Even if you're using native promises AND the value isn't a promise, an `await` ***always*** causes the function to suspend and resume!

You can verify this using this simple example code:
```js
async function test() { console.log(await 1) }

console.log('before');
test();
console.log('after');
```
![image](https://user-images.githubusercontent.com/7462525/171234678-f750beec-e547-473b-90c4-f4d71e23dd89.png)
Without the `await`:
![image](https://user-images.githubusercontent.com/7462525/171234888-e20c88f3-381b-4b72-a088-3fa3ca1d46be.png)

Additionally if you're not targeting a modern browser (eg if you're targeting ES5) than adding an `await` can lead to [a huge amount of JS being generated](https://www.typescriptlang.org/play?target=1#code/PQgEB4CcFMDNpgOwMbVAGwJYCMC8AiaAZ0WgA8AXfUYAPgFgAoJgQyIE8VRYBXFCzAHtEoCsQoAKAJSgA3qGTCig9NAB06QQHMJLAO4tMFUAEYZAXyZNFiZao3aJAcmxxBMJ1IDcTMUUne1koq6po6TiywYpCeXkA) for no reason!

We have a number of strong arguments for why you shouldn't unnecessarily `await`, but I agree - we don't have any arguments for why you should do it.

**Comment by matthieusieben** [comment]
They main reason why using return await is needed is to have proper stack traces ( https://v8.dev/docs/stack-trace-api )

Another reason is to be consistent whether you are in a try/catch and not.
